examples/http-proxy/http_proxy.py

```diff
@@ -0,0 +1,226 @@
+        print(f"{TEXT_COLOR_GREEN}remote content: {data} {TEXT_COLOR_DEFAULT}")
+
+        # todo pass headers to the response ? ...
+        response_text = data.decode("utf-8")
+        print(f"{TEXT_COLOR_GREEN}local response: {response_text}{TEXT_COLOR_DEFAULT}")
+
+    await ws_session.close()
+
+    return web.Response(text=response_text)
+
+
+async def run_local_server(cluster: Cluster, port: int):
+    runner = web.ServerRunner(web.Server(functools.partial(web_root, cluster)))
+    await runner.setup()
+    site = web.TCPSite(runner, port=port)
+    await site.start()
+
+    return site
+
+
+# ######## Golem Service
+
+
+class HttpService(Service):
+    @staticmethod
+    async def get_payload():
+        return await vm.repo(
+            image_hash="de6d4cf997138af384caf32b1d222f35ead1d3639eebafa575359f1d",
+            capabilities=[vm.VM_CAPS_VPN],
+        )
+
+    async def start(self):
+        async for s in super().start():
+            yield s
+
+        s = self._ctx.new_script()
+        s.run("/docker-entrypoint.sh")
+        s.run("/bin/chmod", "a+x", "/")
+        s.run(
+            "/bin/sh",
+            "-c",
+            f"echo running on {shlex.quote(self.provider_name)} >> /usr/share/nginx/html/index.html",
+        )
+        s.run("/usr/sbin/nginx"),
+        yield s
+
+    # we don't need to implement `run` since, after the service is started,
+    # all communication is performed through the VPN
+
+
+# ######## Main application code which spawns the Golem service and the local HTTP server
+
+
+async def main(
+    subnet_tag,
+    driver=None,
+    network=None,
+    num_instances=1,
+    port=8080,
+):
+    async with Golem(
+        budget=1.0,
+        subnet_tag=subnet_tag,
+        driver=driver,
+        network=network,
+    ) as golem:
+
+        print(
+            f"yapapi version: {TEXT_COLOR_YELLOW}{yapapi_version}{TEXT_COLOR_DEFAULT}\n"
+            f"Using subnet: {TEXT_COLOR_YELLOW}{subnet_tag}{TEXT_COLOR_DEFAULT}, "
+            f"payment driver: {TEXT_COLOR_YELLOW}{golem.driver}{TEXT_COLOR_DEFAULT}, "
+            f"and network: {TEXT_COLOR_YELLOW}{golem.network}{TEXT_COLOR_DEFAULT}\n"
+        )
+
+        commissioning_time = datetime.now()
+
+        network = await golem.create_network("192.168.0.1/24")
+        cluster = await golem.run_service(HttpService, network=network, num_instances=num_instances)
+
+        def instances():
+            return [f"{s.provider_name}: {s.state.value}" for s in cluster.instances]
+
+        def still_running():
+            return any([s for s in cluster.instances if s.is_available])
+
+        def still_starting():
+            return len(cluster.instances) < num_instances or any(
+                [s for s in cluster.instances if s.state == ServiceState.starting]
+            )
+
+        # wait until all remote http instances are started
+
+        while still_starting() and datetime.now() < commissioning_time + STARTING_TIMEOUT:
```

Comment:
okay, added an exception if the instances are not started after timeout elapses

Reply:
I don't really like this solution, because we now have more spam (instead of decreased spam amount with `commissioning_time` and `STARTING_TIMEOUT` just removed). But OK, good enough : )